pep-0704.rst

```diff
@@ -0,0 +1,137 @@
+PEP: 704
+Title: Require virtual environments by default for installers
+Author: Pradyun Gedam <pradyunsg@gmail.com>
+Sponsor: Brett Cannon <brett@python.org>
+PEP-Delegate: Paul Moore <p.f.moore@gmail.com>
+Discussions-To: https://discuss.python.org/t/22846
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Content-Type: text/x-rst
+Created: 16-Jan-2023
+Python-Version: 3.13
+Post-History: 16-Jan-2023
+
+
+Abstract
+========
+
+This PEP proposes that package installers like ``pip`` MUST require a virtual environment by default, on Python 3.13+.
+
+Motivation
+==========
+
+Python virtual environments are an essential part of the development workflow for Python. However, it requires extra effort since it is an opt-in feature, and requires users to either:
+
+- take explicit steps to activate/deactivate a virtual environment
+- use ``<path-to-venv>/<bin-path>/<executable>`` to run files
+
+This creates a barrier to entry for many new users, since virtual environments serve as a pre-requisite for typical Python development workflows and how virtual environments work is a lot of information for anyone new. It can take a lot of extra time and effort to explain them.
+
+Further, activation of virtual environment uses a slightly different syntax/mechanism to activate on different platforms.
+
+It also creates a scope for mistakes, since users must remember to activate the virtual environment before running an installer like ``pip``. On certain Linux distributions, this can result in the installer modifying files that are owned by the operating system (this is partially mitigated by :pep:`668` for distributions that opt-in to marking their environments accordingly).
+
+This PEP proposes that the default behaviour of installers like ``pip`` should be to require a virtual environment to be active. This would make it easier for new users to get started with Python, and would reduce the scope for mistakes.
```

Comment:
These two sentences are really Rationale rather than Motivation, as they try to explain why the solution you're proposing should be accepted to solve the problem outlined in the Motivation. Furthermore, I'd strongly suggest expanding that Rationale section at least somewhat so it actually explains how implementing this default behavior will achieve the aim you state here, as currently on the surface it appears to contradict itself (which, to be clear, I believe you have a reasonably plausible case for—you just have to actually make it :slightly_smiling_face: ).

In particular, most of the aforementioned Motivation focuses at some length on the UX challenges that virtual environment present for new users, as it describes the extra steps that using venvs require, states that this "creates a barrier to entry for many new users", and mentions that "how virtual environments work is a lot of information for anyone new. It can take a lot of extra time and effort to explain them."

However, in the Rationale section, it states having installers "require a virtual environment to be active" by default "would make it easier for new users to get started with Python". As the previous lines just established that understanding venvs and knowing the actions to take to create and active one are "a barrier to entry for many new users", and there is no mention here of _how_ this would be made easier for new users, it sounds like a _prima facie_ contradiction that requiring new users to create and activate a venv before installing any package would make it easier, rather than harder, "for new users to get started".

Also, by contrast, the Motivation more briefly touches on the issue of accidentally installing packages without a venv active, mentioning a single concrete example that is at least partially mitigated by another accepted PEP, despite this arguably being the broader and more obvious benefit here.

Therefore, I suggest:

* Make a Rationale section starting with this paragraph.
* Mention how this proposal will actually make thins easier for new users by, e.g.,
  * Mandating that tools give users simple, specific and actionable advice on how to create and activate one, and links to help resources
  * Standardizes the naming of venvs to avoid user confusion and improve tool automation
  * Etc.
* Add at least >=one more example to the Motivation of the real-world problems not activating a venv causes users